fileHub/js-app/app/file-list/file-list-page.js

```diff
@@ -50,62 +49,63 @@ export class FileListPage extends Component {
     const fileList = new FileList(fileListBodyElement);
     fileList.onFolderClick(this._onNavigateToFolder);
 
-    this._stateManager.onStateChanged('locationParams', async (state) => {
+    this._onStateChangedListener('locationParams', async () => {
+      const state = this._stateManager.state;
       const currentFolderId = state.locationParams.currentFolderId;
       if (!currentFolderId && !state.rootFolder) {
         this._stateManager.dispatch(new GetRootFolder());
       } else if (!currentFolderId && state.rootFolder) {
         this._onNavigateToFolder(state.rootFolder.id);
       } else {
-        await this._stateManager.dispatch(new FetchCurrentFolder());
+        this._stateManager.dispatch(new FetchCurrentFolder());
         this._stateManager.dispatch(new FetchCurrentFolderContent());
         if (!state.userData) {
           this._stateManager.dispatch(new GetCurrentUser());
         }
       }
     });
 
-    this._stateManager.onStateChanged('currentFolder', (state) => {
-      breadcrumbs.currentDirectory = state.currentFolder;
+    this._onStateChangedListener('currentFolder', () => {
+      breadcrumbs.currentDirectory = this._stateManager.state.currentFolder;
     });
 
-    this._stateManager.onStateChanged('isCurrentFolderFetching', (state) => {
-      breadcrumbs.loading = state.isCurrentFolderFetching;
+    this._onStateChangedListener('isCurrentFolderFetching', () => {
+      breadcrumbs.loading = this._stateManager.state.isCurrentFolderFetching;
     });
 
-    this._stateManager.onStateChanged('fetchingCurrentFolderErrorMessage', (state) => {
+    this._onStateChangedListener('fetchingCurrentFolderErrorMessage', () => {
       breadcrumbs.currentDirectory = null;
-      breadcrumbs.errorMessage = state.fetchingCurrentFolderErrorMessage;
+      breadcrumbs.errorMessage = this._stateManager.state.fetchingCurrentFolderErrorMessage;
     });
 
-    this._stateManager.onStateChanged('currentFolderContent', (state) => {
-      fileList.fileItems = state.currentFolderContent;
+    this._onStateChangedListener('currentFolderContent', () => {
+      fileList.fileItems = this._stateManager.state.currentFolderContent;
     });
 
-    this._stateManager.onStateChanged('isCurrentFolderContentFetching', (state) => {
+    this._onStateChangedListener('isCurrentFolderContentFetching', () => {
       fileList.fileItems = null;
-      fileList.loading = state.isCurrentFolderContentFetching;
+      fileList.loading = this._stateManager.state.isCurrentFolderContentFetching;
     });
 
-    this._stateManager.onStateChanged('fetchingCurrentFolderContentErrorMessage', (state) => {
+    this._onStateChangedListener('fetchingCurrentFolderContentErrorMessage', () => {
       fileList.fileItems = null;
-      fileList.errorMessage = state.fetchingCurrentFolderContentErrorMessage;
+      fileList.errorMessage = this._stateManager.state.fetchingCurrentFolderContentErrorMessage;
```

Comment:
Looks like having state in listener properties was much easier. Why was this changed?